corehq/middleware.py

```diff
@@ -90,14 +93,44 @@ def process_response(self, request, response):
 
 class TimeoutMiddleware(object):
 
-    def process_request(self, request):
+    @staticmethod
+    def _session_expired(timeout, activity, time):
+        if activity is None:
+            return False
+        if time - string_to_utc_datetime(activity) > datetime.timedelta(minutes=timeout):
+            return True
+        else:
+            return False
+
+    @staticmethod
+    @quickcache(['domain_name', 'secure_session'], timeout=10 * 60)
+    def _secure_domain_not_session(secure_session, domain_name):
+        if secure_session:
+            return False
+        else:
+            return Domain.get_by_name(domain_name).secure_sessions
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
```

Comment:
changed from process_request to process_view because the Dimagi middleware that adds a request.domain attribute is a process view function.